src/libsystemd/sd-daemon/sd-daemon.c

```diff
@@ -595,6 +595,17 @@ _public_ int sd_pid_notify_with_fds(
         r = -errno;
 
 finish:
+        if (DEBUG_LOGGING && r < 0) {
+                _cleanup_free_ char *c = NULL;
+
+                c = strdup(state);
+                if (c)
+                        string_replace_char(c, '\n', ' ');
```

Comment:
OOM check?

Reply:
We don't want to lose the original error since this is purely debugging stuff